### PR TITLE
data(key_indicators): garden and grapher

### PR DESCRIPTION
**Pull request by lucasrodes**
**Key Indicators** dataset is now available in Grapher. Find below the complete list of updates from this PR:

- Metric `Population Density` is now included within the **Key Indicators** Garden dataset. The old dataset has been deprecated.
- Metadata for **Key Indicators** has been restructured in a single YAML file (Garden step).
- Implemented Grapher step to import **Key Indicators** to Grapher.

### CODE REVIEW

**Review by pabloarosado — APPROVED** [review]
Nice. I've just made a few comments, feel free to disregard them. @Marigold and I were just talking in the tech tea that in principle it should be possible to specify `sources` as a separate entity (like `dataset` or `tables`) inside the yaml file, so that you don't have to repeat the source descriptions every time for each variable. But I don't know if it's straightforward.

File: etl/steps/data/garden/owid/latest/key_indicators/key_indicators.meta.yml
Comment (lucasrodes) [reply etl/steps/data/garden/owid/latest/key_indicators/key_indicators.meta.yml:98]: Thanks!

File: etl/steps/data/garden/owid/latest/key_indicators/key_indicators.meta.yml
Comment (lucasrodes) [reply etl/steps/data/garden/owid/latest/key_indicators/key_indicators.meta.yml:6]: Thank!

File: etl/steps/data/garden/owid/latest/key_indicators/table_population_density.py
Comment (lucasrodes) [reply etl/steps/data/garden/owid/latest/key_indicators/table_population_density.py:63]: Thing is most of the description can be reused, so I decided to do it like this

File: etl/steps/grapher/owid/latest/key_indicators.py
Comment (lucasrodes) [reply etl/steps/grapher/owid/latest/key_indicators.py:44]: Yup, cool, thanks @pabloarosado!

For the moment, I am happy to use this patch. Once, the fix is done, I'll remove this patch.

**Comment by Marigold** [comment]
@pabloarosado is right, you can use yaml anchors to DRY your yaml file, e.g.

```yml
tables:
  population:
    title: Population (Gapminder, HYDE & UN)
    description: ...
    variables:
      population:
        title: Population
        sources: &population-sources
          - name: Gapminder (v6)
            published_by: Gapminder (v6)
            url: https://www.gapminder.org/data/documentation/gd003/
            date_accessed: October 8, 2021
          - name: UN (2022)
            published_by: United Nations - Population Division (2022)
            url: https://population.un.org/wpp/Download/Standard/Population/
            date_accessed: September 10, 2022
          - name: HYDE (v3.2)
            published_by: HYDE (v3.2)
            url: https://dataportaal.pbl.nl/downloads/HYDE/
            date_accessed: October 8, 2021
      world_pop_share:
        title: Share of world population
        sources: *population-sources
```

**Comment by pabloarosado** [comment]
> @pabloarosado is right, you can use yaml anchors to DRY your yaml file, e.g.
> 
> ```yaml
> tables:
>   population:
>     title: Population (Gapminder, HYDE & UN)
>     description: ...
>     variables:
>       population:
>         title: Population
>         sources: &population-sources
>           - name: Gapminder (v6)
>             published_by: Gapminder (v6)
>             url: https://www.gapminder.org/data/documentation/gd003/
>             date_accessed: October 8, 2021
>           - name: UN (2022)
>             published_by: United Nations - Population Division (2022)
>             url: https://population.un.org/wpp/Download/Standard/Population/
>             date_accessed: September 10, 2022
>           - name: HYDE (v3.2)
>             published_by: HYDE (v3.2)
>             url: https://dataportaal.pbl.nl/downloads/HYDE/
>             date_accessed: October 8, 2021
>       world_pop_share:
>         title: Share of world population
>         sources: *population-sources
> ```

Ah nice. This is not exactly the idea I had in mind but it's also good. What I thought was having:
```yml
sources:
  population_source_1:
    name: Gapminder...
  population_source_2:
    name: ...
    ...

tables:
  variables:
    population:
      sources: *sources:population_source_1 *sources:population_source_2
```
Or something like that (alternatively, we could also group together `population_sources`, to avoid repetitions). Would this be possible? I just thought that it may be clearer to specify the sources in a separate section. Otherwise your solution is good.

**Comment by lucasrodes** [comment]
I am getting some issues with 

```python
# Add sources from variables (ensure sources are not duplicated)
ds.metadata.sources = [dict(ss) for ss in set(frozenset(s.items()) for s in sources)]
as the type of sources has changed from List[dict] to List[Source]
```

(from file `etl/steps/data/garden/owid/latest/key_indicators/__init__.py`)

Probably due to the implementation change from https://github.com/owid/etl/issues/486.

Working on a fix

**Comment by lucasrodes** [comment]
Done, grapher step working

File: etl/steps/grapher/owid/latest/key_indicators.py
Comment (Marigold) [inline etl/steps/grapher/owid/latest/key_indicators.py:44]: @pabloarosado is this something we can do for all datasets in `combine_metadata_sources` or do you have any objections?

**Review by Marigold — APPROVED** [review]
Looks good!

File: etl/steps/grapher/owid/latest/key_indicators.py
Comment (Marigold) [reply etl/steps/grapher/owid/latest/key_indicators.py:44]: @lucasrodes since this just got merged, could you remove those `_patch` functions and then merge your PR? (I have another PR that is waiting for this one)

Comment (lucasrodes) [reply etl/steps/grapher/owid/latest/key_indicators.py:44]: On it! Will ping you

**Comment by Marigold** [comment]
@lucasrodes could you please also squash your commits before merging? (choose "Squash and merge" in the button for merging)

**Comment by larsyencken** [comment]
Hi friends! It looks like this one passed review, but perhaps it didn't get executed fully, neither by Lucas or by reviewers? Afterwards, `etl key_indicators` failed, not because of any issue in this dataset, but because of how other steps were referencing it.

This has happened in a bunch of PRs recently, so comment is not even about this change specifically. We need CI badly for the ETL, but in the meantime, please run the step and all its downstream dependencies to confirm that it all works before approving a PR and once more before merging 🙏

Next #data-architecture chat on Thu, let's also try to work out how we can make all this more reliable.

**Comment by lucasrodes** [comment]
@larsyencken Got it!

Should we then take a look at the dag files, and run `etl` for all datasets that depend on the one that got modified (in this case 'key indicators')?

**Comment by larsyencken** [comment]
@lucasrodes Not to worry, Pablo shipped this fix on Friday: https://github.com/owid/etl/pull/506

Mainly I just wanted to flag that we need a way of ensuring the ETL works before merging. I have a draft CI check that I'm trying out, if it works it might solve this problem.

**Comment by lucasrodes** [comment]
thanks @larsyencken, happy to test when it is ready

**Pull request merged**
